StripeConnect/StripeConnectTests/Internal/Webview/ConnectComponentWebViewTests.swift

```diff
@@ -13,30 +13,96 @@ import WebKit
 import SafariServices
 
 class ConnectComponentWebViewTests: XCTestCase {
-    func testSetup() {
+    @MainActor
+    func testFetchClientSecret() async throws {
+        let expectation = XCTestExpectation(description: "Client secret is fetched")
         let componentManager = EmbeddedComponentManager(apiClient: .init(publishableKey: "test"), fetchClientSecret: {
-            XCTFail("Client secret should not be retrieved in this test")
+            expectation.fulfill()
             return ""
         })
-        let webView = ConnectComponentWebView(componentManager: componentManager, componentType: .payouts)
+        let webView = ConnectComponentWebView(componentManager: componentManager, componentType: .payouts, loadContent: false)
         
-        XCTAssertEqual(webView.url, URL(string: "https://connect-js.stripe.com/v1.0/ios_webview.html#component=payouts&publicKey=test"))
+        try await webView.evaluateMessageWithReply(name: "fetchClientSecret",
+                                                   json: "{}")
+        
+        await fulfillment(of: [expectation], timeout: 1.0)
     }
     
     @MainActor
-    func testFetchClientSecret() async throws {
-        let expectation = XCTestExpectation(description: "Client secret is fetched")
+    func testFetchInitParams() async throws {
+        let expectation = self.expectation(description: "Message received")
+        let message = FetchInitParamsMessageHandler.Reply(locale: "fr-FR")
         let componentManager = EmbeddedComponentManager(apiClient: .init(publishableKey: "test"), fetchClientSecret: {
-            expectation.fulfill()
+            XCTFail("Client secret should not be retrieved in this test")
             return ""
         })
-        let webView = ConnectComponentWebView(componentManager: componentManager, componentType: .payouts)
+        let webView = ConnectComponentWebView(componentManager: componentManager, 
+                                              componentType: .payouts,
+                                              webLocale: Locale(identifier: "fr_FR"),
+                                              loadContent: false)
+       
         
-        try await webView.evaluateMessageWithReply(name: "fetchClientSecret",
+        try await webView.evaluateMessageWithReply(name: "fetchInitParams",
                                                    json: "{}",
-                                                   postReply: false)
+                                                   verifyResult: { result in
+            guard let json = try? JSONEncoder.connectEncoder.encode(message) else {
+                XCTFail("Failed to encode message \(message)")
+                return
+            }
+            XCTAssertEqual(String(data:result, encoding: .utf8), String(data:json, encoding: .utf8))
```

Comment:
Why does this need to be encoded to be compared? Could we instead directly compare `result` and `message`?

EDIT: I see now that the type of `evaluateMessageWithReply` changed to send a raw `Data` type instead of the payload type. This makes the function a lot more challenging to read. If the intent is to compare the encoded JSON string, then I think it would be better if the test helper returned the String directly instead of the data type and then having to decode it in the test.

So ideally, this would read something like:
```
try await webView.evaluateMessageWithReply(name: "fetchInitParams",
                                           json: "{}",
                                           verifyResult: { result in
    XCTAssertEqual(result, FetchInitParamsMessageHandler.Reply(locale: "fr-FR"))
}
```
...or this (if it's necessary to test the JSON encoding here as well):
```
    XCTAssertEqual(result, "{\"locale\":\"fr-FR\"}")
```